src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -743,14 +744,40 @@ internal static bool IsPackageReferenceProject(Project project)
                     if (matchingPackages.Any())
                     {
                         var topLevelPackage = matchingPackages.Single();
-                        InstalledPackageReference installedPackage;
+                        InstalledPackageReference installedPackage = default;
 
                         //If the package is not auto-referenced, get the version from the project file. Otherwise fall back on the assets file
                         if (!topLevelPackage.AutoReferenced)
                         {
                             try
                             { // In case proj and assets file are not in sync and some refs were deleted
-                                installedPackage = projPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+                                var projectPackage = projectPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+
+                                // If the project is using CPM and it's not using VersionOverride, get the version from Directory.Package.props file
+                                if (assetsFile.PackageSpec.RestoreMetadata.CentralPackageVersionsEnabled && !projectPackage.IsVersionOverride)
+                                {
+                                    if (topLevelPackage.VersionOverride?.OriginalString is not null)
+                                    {
+                                        installedPackage = new InstalledPackageReference(topLevelPackage.Name)
+                                        {
+                                            OriginalRequestedVersion = topLevelPackage.VersionOverride?.OriginalString,
+                                        };
+                                    }
+                                    else
+                                    {
+                                        ProjectRootElement directoryBuildPropsRootElement = GetDirectoryBuildPropsRootElement(project);
+                                        ProjectItemElement packageInCPM = directoryBuildPropsRootElement.Items.Where(i => (i.ItemType == PACKAGE_VERSION_TYPE_TAG || i.ItemType.Equals("GlobalPackageReference")) && i.Include.Equals(topLevelPackage.Name, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+
+                                        installedPackage = new InstalledPackageReference(topLevelPackage.Name)
+                                        {
+                                            OriginalRequestedVersion = packageInCPM.Metadata.FirstOrDefault(i => i.Name.Equals("Version", StringComparison.OrdinalIgnoreCase)).Value,
+                                        };
+                                    }
+                                }
+                                else
+                                {
```

Comment:
This gets the version from versionoverride right?

Reply:
FYI, I use github's website to look at PR comments, but I believe that you use VSCode. This is what I see: 
![image](https://github.com/NuGet/NuGet.Client/assets/5030577/e8c14169-fa33-4566-a7ec-77dca373c3ea)

I assume you're referring to the code UNDER the line the comment is on, but github shows the 4 lines of code above. Your comment doesn't make sense with the info that github provides, so I'd like to ask you to please adapt to how github presents PR comments, so that it makes more sense for reviewers.

Reply:
I was referring to the code below yeah. 

I'll try to remember that for the future.